module/zfs/zap.c

```diff
@@ -545,6 +545,13 @@ zap_get_leaf_byblk(zap_t *zap, uint64_t blkid, dmu_tx_t *tx, krw_t lt,
 	ASSERT3U(db->db_size, ==, 1 << bs);
 	ASSERT(blkid != 0);
 
+	zap_leaf_phys_t *zap_phys = db->db_data;
+	if (zap_phys->l_hdr.lh_block_type != ZBT_LEAF ||
+	    zap_phys->l_hdr.lh_magic != ZAP_LEAF_MAGIC) {
+		dmu_buf_rele(db, NULL);
+		return (SET_ERROR(EIO));
+	}
```

Comment:
No objection, but this looks unrelated.  Is this something you just happened to hit while testing?

Reply:
I haven't noticed any issue with that. And this was part of the original PR (I can revert it and run some tests if you want).